Hybrid-state.md

```diff
@@ -0,0 +1,93 @@
+## Table of contents
+
+* [Overview](#overview)
+* [End-to-End tests migration tracker](#end-to-end-tests-migration-tracker)
+* [Add or Modify E2E tests](#add-or-modify-e2e-tests)
+  * [If contributor wants to add a new suite](#if-contributor-wants-to-add-a-new-suite)
+  * [If contributor wants to modify a suite](#if-contributor-wants-to-modify-a-suite)
+  * [If contributor wants to make some changes to the common files](#if-contributor-wants-to-make-some-changes-to-the-common-files)
+  * [Current approach to migrate the dependencies](#current-approach-to-migrate-the-dependencies)
+* [Contact](#contact)
+
+## Overview
+
+Oppia currently uses **Protractor** for end-to-end testing, but as the Angular team plans to end the development of Protractor at the end of 2022, we are planning to migrate all these tests to **WebdriverIO**.
+
+As long as the migration is going on, the tests will be present in the **Hybrid State**, i.e. the tests will be present in both webdriverIO and protractor.
+
+There are a few important points about the hybrid state:
+
+* We will have two tools to run the E2E tests: protractor and webdriverIO.
+
+* Each test suite will be either present in webdriverIO or protractor, never in both.
+
+* When a utility file is needed by both a protractor test and a webdriverIO test, we will have two copies of the file: one in webdriverIO and one in protractor.
+
+* No file will contain the code of both webdriverIO and protractor.
+
+* Protractor suite can only work on protractor utils and vice versa for webdriverIO.
+
+## End-to-End tests migration tracker
+
+The [webdriverIO migration tracker](https://docs.google.com/spreadsheets/d/1Mj-llYXMURtis54vpL2VL7BwgRiFIZ1nIFtK3fY3Se4/edit?usp=sharing) contains the following details:
+
+* Suites migrated to webdriverIO.
+
+* Suites still in protractor.
+
+* Expected date of migration of protractor suites.
+
+* Dependencies migrated to webdriverIO.
+
+* Dependencies still in protractor.
+
+* List of common dependencies which are present in both protractor and webdriverIO.
+
+Contributors can use this to get the idea about when a test suite is going to be migrated and hence plan their PR accordingly in order to avoid clashes which have been discussed in [Add or Modify E2E tests](#add-or-modify-e2e-tests) section (See the [edge case in point 2](#if-contributor-wants-to-modify-a-suite) for more information in next section).
+
+## Add or modify E2E tests
+
+#### If contributor wants to add a new suite
+
+  Add the new e2e test suite in webdriverIO.
+
+  **Note**: As the first new suite will be added in late August, we will have most of the files already migrated to WebdriverIO like action.js, forms.js, user.js, general.js, and waitFor.js which are mostly needed for writing a new test.
```

Comment:
Actually, it was not getting highlighted, so I thought the comment was just for the line gap. Done now.

Reply:
For future reference, GitHub only highlights words in the diff when it can figure out a word-level diff. When it can't figure that out, nothing will be highlighted even though there are changes to make